src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java

```diff
@@ -2187,6 +1982,336 @@ public static boolean areAnyGencodeFuncotation(final List<Funcotation> funcotati
+                altAaSeq = alternateProteinSequence.substring(proteinChangeStartIndex, proteinChangeStartIndex + numAltAminoAcids );
+            }
+            else {
+                // To start with, we fill in the information naively corresponding to the potentially
+                // changed amino acid sequence:
+                proteinChangeStartIndex = ((alignedCodingSequenceAlleleStart-1) / 3);
+
+                aaStartPos = proteinChangeStartIndex + 1;
+                aaEndPos = aaStartPos + numRefAminoAcids;
+
+                refAaSeq = referenceProteinSequence.substring(proteinChangeStartIndex, aaEndPos);
+                altAaSeq = alternateProteinSequence.substring(proteinChangeStartIndex, aaStartPos + numAltAminoAcids);
+
+                // OK, now we have a naive, but correct protein change.
+                // We should attempt to simplify it into a simple insertion.
+                // To do this we must detect the bases that have been inserted into the reference amino acid
+                // string.
+                // For this case, we check the first and last amino acids in the ref and alt strings to
+                // see if they match.  If so we remove that amino acid from the change string.
+                //
+                // Note this case is similar to the complementary deletion case.
+                // Note that we must attempte to alternate looking at the front and the back to get the proper
+                //      and correct protein change.
+                boolean frontMustBeTrimmed = (!refAaSeq.isEmpty()) && (!altAaSeq.isEmpty()) && (refAaSeq.charAt(0) == altAaSeq.charAt(0));
+                boolean backMustBeTrimmed  = true;
+
+                while ( frontMustBeTrimmed || backMustBeTrimmed ) {
+                    if ( frontMustBeTrimmed ) {
+                        aaEndPos++;
+                        refAaSeq = refAaSeq.substring(1);
+                        altAaSeq = altAaSeq.substring(1);
+                    }
+
+                    backMustBeTrimmed = (!refAaSeq.isEmpty()) && (refAaSeq.charAt(refAaSeq.length() - 1) == altAaSeq.charAt(altAaSeq.length() - 1));
+                    if ( backMustBeTrimmed ) {
+
+                        // Must adjust start position so that insertions can occur with a correct range:
+                        if ( aaStartPos == aaEndPos ) {
+                            --aaStartPos;
+                        }
+                        else {
+                            --aaEndPos;
+                        }
+
+                        refAaSeq = refAaSeq.substring(0, refAaSeq.length() - 1);
+                        altAaSeq = altAaSeq.substring(0, altAaSeq.length() - 1);
+                    }
+
+                    frontMustBeTrimmed = (!refAaSeq.isEmpty()) && (!altAaSeq.isEmpty()) && (refAaSeq.charAt(0) == altAaSeq.charAt(0));
+                }
+            }
+        }
+        else if ( GATKVariantContextUtils.isDeletion( refAllele, altAllele ) ) {
```

Comment:
I've refactored the code to be a little easier to read.

At this point it would be difficult to factor out the shared code (specifically for the trimming / reduction) because of the small differences in operations.